### DISCUSSION TITLE
Upsert in storage

### DISCUSSION THREAD

**Discussion by nextdev1111** [supabase/supabase#7733] [Questions] [answered]
How can I update the existing path in supabase storage. And if the path doesn't exists then it would create it.

**Comment by GaryAustin1** [comment]
Have you tried setting the upsert flag to true for upload or update?   I've not tried it, but believe that is what it is for.
![image](https://user-images.githubusercontent.com/54564956/179403423-8c77602e-1d84-4c46-a601-6b122b166bd8.png)

**Reply by nextdev1111** [reply]
No, it doesn't work.

**Reply by obrunsmann** [reply]
There was one little thing that held me up a bit: It's important to grant SELECT, UPDATE and INSERT to the user. (I was missing the SELECT .... )

**Comment by GaryAustin1** [comment, marked as answer]
What part does not work?  Do you get an error?  It does not create a new file, update the old file?
Remember if you update an existing file, you will still possibly get the old file data until the caches have expired in the browser and cdn.

**Reply by nextdev1111** [reply]
I updated the file and got no error but when i check the sulbase storage, I can only get the old one

**Reply by GaryAustin1** [reply]
After your cache time(default 1 hour) your new file should show up.  You can also add a cache buster string to the path like ?t=4748838. The number can be random or the time.   The SB dashboard uses this method to get the current file.  Only do this though were you must see the results of a change immediately (like an admin panel).

**Reply by nextdev1111** [reply]
Ok
